src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,20 +215,22 @@ public static String formatTime12H(Date date) {
         return sdf.format(date);
     }
 
-    public static String formatDateTimeForComments(Date date) {
+    public static String formatDateTimeForComments(Date date, double sessionTimeZone) {
         if (date == null) {
             return "";
         }
         SimpleDateFormat sdf = null;
         Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
+        TimeZone timeZone = getTimeZoneFromDoubleOffset(sessionTimeZone);
+        c.setTimeZone(timeZone);
         c.setTime(date);
+        sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm a Z");
+        sdf.setTimeZone(timeZone);
         if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
-            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm");
-            sdf.setTimeZone(SystemParams.TIME_ZONE);
-            return sdf.format(date) + " NOON UTC";
+            sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm Z");
+            sdf.setTimeZone(timeZone);
+            return sdf.format(date) + " NOON";
         }
-        sdf = new SimpleDateFormat("EEE, dd MMM yyyy, hh:mm a zzz");
-        sdf.setTimeZone(SystemParams.TIME_ZONE);
```

Comment:
Sorry, I did not notice that the two patterns were different. You may want to revert the change to move the related lines up.